src/scitokens/scitokens.py

```diff
@@ -155,22 +212,29 @@ def deserialize(serialized_token, require_key=True):
 
         Verifies the claims pass the current set of validation scripts.
         """
-        info = serialized_token.split(".")
+        info = serialized_token.decode('utf8').split(".")
         if len(info) != 3 and len(info) != 4: # header, format, signature[, key]
             raise MissingKeyException("No key present in serialized token")
 
         key = info[-1]
         serialized_jwt = info[0] + "." + info[1] + "." + info[2]
 
         unverified_headers = jwt.get_unverified_header(serialized_jwt)
-        unverified_payload = json.loads(decode_base64(info[1]))
+        unverified_payload = json.loads(decode_base64(info[1].encode("ascii")).decode('utf-8'))
         
         # Get the public key from the issuer
         issuer_public_key = SciToken._get_issuer_publickey(unverified_headers, unverified_payload)
         
         claims = jwt.decode(serialized_token, issuer_public_key)
-        print(claims)
-        return
+        # Do we have the private key?
+        if (len(info) == 4):
+            to_return = SciToken(serialized_token = serialized_token, key = key)
```

Comment:
Create the token, then set the member variable directly.  Let's not add `serialized_token` to the constructor API.